### PR TITLE
Fixed #35117 -- Added support for the hectare unit in Area.

### PR DESCRIPTION
**Pull request by alexisig**
Forum post : https://forum.djangoproject.com/t/geodjango-new-measure-unit-suggestion-hectare/26923/3
Ticket: https://code.djangoproject.com/ticket/35117

### CODE REVIEW

**Comment by felixxm** [comment]
@alexisig Thanks for this patch :+1: All new features require [an accepted ticket](https://code.djangoproject.com/newticket), please create one (check out e.g. ticket-30341).

**Comment by alexisig** [comment]
@felixxm Thank you for the pointer!

File: django/contrib/gis/measure.py [hunk 1]
```diff
@@ -346,9 +346,16 @@ def __mul__(self, other):
 
 class Area(MeasureBase):
     STANDARD_UNIT = AREA_PREFIX + Distance.STANDARD_UNIT
+    AREA_ONLY_UNITS = {"ha": 10000}
+    AREA_ONLY_ALIAS = {"hectare": "ha"}
```
Comment (felixxm) [inline django/contrib/gis/measure.py:350]: I don't think we need a separate variables for area-only units.

File: tests/gis_tests/test_measure.py [hunk 2]
```diff
@@ -38,6 +38,11 @@ def test_init_invalid(self):
         with self.assertRaises(AttributeError):
             D(banana=100)
 
+    def test_init_invalid_area_only_units(self):
+        "Testing initialization from area only units"
```
Comment (felixxm) [inline tests/gis_tests/test_measure.py:42]: I know you treated `test_init_invalid` as a pattern, but this docstring is unnecessary. Please remove it.

File: tests/gis_tests/test_measure.py [hunk 3]
```diff
@@ -294,6 +299,19 @@ def test_units_str(self):
         self.assertEqual(repr(a1), "Area(sq_m=100.0)")
         self.assertEqual(repr(a2), "Area(sq_km=3.5)")
 
+    def test_hectare(self):
+        a = A(sq_m=10000)
+        self.assertEqual(a.ha, 1)
+
+    def test_area_only_unit_att_name(self):
+        "Testing the `unit_attname` class method"
```
Comment (felixxm) [inline tests/gis_tests/test_measure.py:307]:
```suggestion
```

File: tests/gis_tests/test_measure.py [hunk 4]
```diff
@@ -294,6 +299,19 @@ def test_units_str(self):
...
+        unit_tuple = [
+            ("Hectare", "ha"),
+        ]
+        for nm, att in unit_tuple:
+            with self.subTest(nm=nm):
+                self.assertEqual(att, A.unit_attname(nm))
```
Comment (felixxm) [inline tests/gis_tests/test_measure.py:313]: We have only one area-only unit, so we don't need a generic test:
```suggestion
        self.assertEqual(A.unit_attname("Hectare"), "ha")
```

File: docs/ref/contrib/gis/measure.txt [hunk 5]
```diff
@@ -112,7 +113,8 @@ Unit Attribute                     Full name or alias(es)
 .. note::
 
     :class:`Area` attributes are the same as :class:`Distance` attributes,
-    except they are prefixed with ``sq_`` (area units are square in nature).
+    except they are prefixed with ``sq_`` (for square), with the exception
+    of area only units (i.e. hectare)
```
Comment (kezabelle) [inline docs/ref/contrib/gis/measure.txt:117]: Missing trailing full-stop `.`

```suggestion
    of area only units (i.e. hectare).
```

File: docs/ref/contrib/gis/measure.txt [hunk 6]
```diff
@@ -97,6 +97,7 @@ Unit Attribute                     Full name or alias(es)
 ``british_chain_benoit``           British chain (Benoit 1895 B)
 ``british_chain_sears_truncated``  British chain (Sears 1922 truncated)
 ``gold_coast_ft``                  Gold Coast foot
+``ha``                             Hectare
```
Comment (felixxm) [inline docs/ref/contrib/gis/measure.txt:100]: I think it's better to move it to the separate table.

**Review by felixxm — COMMENTED** [review]
@alexisig Thanks :+1: Welcome aboard :boat: 

I pushed small edits to docs.

**Comment by felixxm** [comment]
@alexisig Great timing! You're Django contributor no. 2500!

https://fosstodon.org/@felixxm/111769552802139711

**Pull request merged**